src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Listings/Services/ListingService.cs

```diff
@@ -17,9 +17,16 @@ public class ListingService(
     : IListingService 
 {
     public IQueryable<Listing> Get(
-        Expression<Func<Listing, bool>>? predicate = default,
+        FilterPagination filterPagination,
         bool asNoTracking = false)
-        => listingRepository.Get(predicate, asNoTracking);
+    {
+        return listingRepository.Get(asNoTracking: asNoTracking)
+            .Include(listing => listing.ImagesStorageFile
+                .OrderBy(image => image.OrderNumber))
+            .ThenInclude(media => media.StorageFile)
+            .Skip((int)((filterPagination.PageToken - 1) * filterPagination.PageSize))
+            .Take((int)filterPagination.PageSize);
```

Comment:
You can use apply pagination extension

Reply:
Apply pagination extension method is only available with query specifications for now. I think it should have been fixed in #287 issue.